### PR TITLE
Ignore the --no-debug option when creating the console

### PR DESCRIPTION
**Pull request by aschempp**
Correctly fixes https://github.com/contao/contao/issues/617

### CODE REVIEW

**Review by Toflar — APPROVED** [review]

**Review by leofeyer — COMMENTED** [review]
I do not like the longish method names `createFromConsoleInput()` and `createFromRequest()`. Can we have one `ContaoKernel::create()` method that accepts either a request object or an input object and handles both accordingly?

**Comment by aschempp** [comment]
I don‘t think thats a good idea. There would be no type hinting and no IDE support?

**Comment by aschempp** [comment]
Btw. we could shorten the names to `fromInput` and `fromRequest`. I see that Symfony does similarly: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Process/Process.php#L191

**Comment by leofeyer** [comment]
> Btw. we could shorten the names to `fromInput` and `fromRequest`.

I like this. 👍

**Comment by aschempp** [comment]
Changed in 5efb653bb808a984b25f2ddc3224a3c693ab67bc

File: manager-bundle/src/Console/ContaoApplication.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public static function createConsoleInput()
+    {
+        $argv = $_SERVER['argv'];
+
+        // Ignore the --no-debug option
+        foreach ($argv as $k => $v) {
+            if ($v === '--no-debug') {
+                unset($argv[$k]);
```
Comment (leofeyer) [inline manager-bundle/src/Console/ContaoApplication.php]: This means that if someone passes the `--no-debug` option, they will not be notified that the option does not exist, right? Not sure if this is good.

Comment (aschempp) [reply manager-bundle/src/Console/ContaoApplication.php]: That's correct. Do you think we should show a console warning?

Comment (leofeyer) [reply manager-bundle/src/Console/ContaoApplication.php]: No, I think we should remove the `createConsoleInput()` method again, so the regular exception is thrown. I realize that this means that we cannot use the Symfony Deployer receipt without adjustment anymore, but after our discussion a couple of days ago, I came to the conclusion that a Contao Deployer receipt would solve this way better. 🙄

Comment (aschempp) [reply manager-bundle/src/Console/ContaoApplication.php]: Well we can do that, but not the fromInput method but the one where the input is created. I‘ll adjust it tomorrow

Comment (leofeyer) [reply manager-bundle/src/Console/ContaoApplication.php]: The `fromInput()` method does not need to be adjusted. We just have to replace `$input = ContaoApplication::createConsoleInput();` with `$input = new ArgvInput();` again in the `contao-console` file.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**